### PR TITLE
Update quickstart tutorial

### LINKED ISSUE #462: Update the quickstart in the docs
The quickstart is currently out of date. We should remove the STAC_IO references completely (HTTP reading is now supported out of the box, has been for a bit), uses the old extension methods, and need to point to a new catalog - the current example is 0.6.

### LINKED ISSUE #606: Quickstart and Tutorial documentation out of date
Hi there, new to STAC and trying to learn but hitting some roadblocks with the documentation.

 I can’t get the quickstart tutorials to run or produce the same outputs that are in the documentation - is the documentation up to date? 
 
Some lines of code gave me back errors or the meta data returned was blank when the example showed a different result for the same URL. I cloned the repo and installed it directly and still had similar issues.

I have stactools 0.2.1 and pystac 1.1.0 python 3.8.8 on MacOS.

```
#this doesn't work

from urllib.parse import urlparse
import requests
from pystac import STAC_IO

def requests_read_method(uri):
    parsed = urlparse(uri)
    if parsed.scheme.startswith('http'):
        return requests.get(uri).text
    else:
        return STAC_IO.default_read_text_method(uri)
```

```
STAC_IO.read_text_method = requests_read_method
ImportError: cannot import name 'STAC_IO' from 'pystac' (/opt/anaconda3/lib/python3.8/site-packages/pystac/__init__.py)
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-8f6ab85a192b> in <module>
      3 from urllib.parse import urlparse
      4 import requests
----> 5 from pystac import STAC_IO
      6 
      7 def requests_read_method(uri):

ImportError: cannot import name 'STAC_IO' from 'pystac' (/opt/anaconda3/lib/python3.8/site-packages/pystac/__init__.py)

```

It looks like `StacIO` exists in __init__ but not `STAC_IO`

Additionally, the jupyter notebooks aren't reproducible e.g. `cat = pystac.Catalog.from_file('example-catalog/catalog.json')
cat.describe()` does not yield that same results in the docs.

#462

### PR DESCRIPTION
**Pull request by duckontheweb**
**Related Issue(s):**
- Closes #606
- Closes #462
- #665

**Description:**

This PR updates the Quickstart tutorial notebook to be up-to-date with the current version of PySTAC (v1.2.0). It also uses an example catalog that is included in the repository instead of depending on an external catalog. This will help make the tutorial more self-contained, and should make it easier to ensure that the example we are using is in sync with the STAC spec version that PySTAC currently supports.

**PR Checklist:**

- [x] Code is formatted (run `pre-commit run --all-files`)
- [x] Tests pass (run `scripts/test`)
- [x] Documentation has been updated to reflect changes, if applicable
- [x] This PR maintains or improves overall codebase code coverage.
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/main/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

**Comment by duckontheweb** [comment]
@KennSmithDS It would be good to get your feedback on whether this clarifies some of the issues that you ran into when getting started.

File: docs/quickstart.ipynb [hunk 1]
```diff
@@ -131,496 +146,674 @@
...
+      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/var/folders/td/xzlhhvqn6gn7_zmrjc0zfb6c0000gn/T/tmpiza4ckl4'"
      ]
     }
    ],
    "source": [
-    "item = next(local_root_2.get_all_items())\n",
-    "item.bbox = ['Not a valid bbox']\n",
-    "item.validate()"
+    "shutil.rmtree(tmp_dir)"
```
Comment (lossyrob) [inline docs/quickstart.ipynb:816]: Should use the `ignore_errors=True` to avoid error output

File: docs/quickstart.ipynb [hunk 2]
```diff
@@ -6,68 +6,79 @@
...
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "First, we want to hook into PySTAC to allow for reading of HTTP STAC items, as described in [the STAC_IO Concepts docs](concepts.html#using-stac-io). \n",
+    "## Reading a Catalog\n",
+    "\n",
+    "[A STAC Catalog](https://github.com/radiantearth/stac-spec/tree/master/catalog-spec) is used to group other STAC objects like Items, Collections, or even other Catalogs.\n",
+    "\n",
+    "We will be using a small example catalog adapted from the `geotrellis` [example Landsat Collection](https://github.com/geotrellis/geotrellis-server/tree/977bad7a64c409341479c281c8c72222008861fd/stac-example/catalog/landsat-stac-collection). All STAC Items and Collections can be found in the [docs/example-catalog](./example-catalog) directory of this repo; all Assets are hosted in the Landsat S3 bucket.\n",
```
Comment (lossyrob) [inline docs/quickstart.ipynb:24]: Should capitalize project name `GeoTrellis`. Also perhaps link to project homepage (https://geotrellis.io/)

File: docs/quickstart.ipynb [hunk 3]
```diff
@@ -79,24 +90,23 @@
...
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We can print some information about the catalog, including how many children it has:"
+    "## Crawling Child Catalogs/Collections\n",
+    "\n",
+    "[STAC Collections](https://github.com/radiantearth/stac-spec/tree/master/collection-spec) are used to group related Items and provide aggregate or summary metadata for those Items.\n",
+    "\n",
+    "STAC Catalogs may have many nested layers of Catalogs or Collections within the top-level collection. Our example catalog has one Collection within the main Catalog at ([landsat-8-l1/collection.json](./example-catalog/landsat-8-l1/collection.json). We can list the Collections in a given Catalog using the [`Catalog.get_collections`](https://pystac.readthedocs.io/en/latest/api.html#pystac.Catalog.get_collections) method. This method returns an iterable of PySTAC [`Collection`](https://pystac.readthedocs.io/en/latest/api.html#collection) instances, which we will turn into a `list`."
```
Comment (lossyrob) [inline docs/quickstart.ipynb:109]: 2 things:
- There's a stray open paren in the first landsat link that throws off rendering.
- The nbsphinx can't handle backticks in the brackets for links (seen when rendering the quickstart through `make livehtml` in the docs folder). You should remove backticks, e.g. [\`Catalog.get_collections\`] ->[Catalog.get_collections]

**Review by lossyrob — APPROVED** [review]
Content looks great.

Made some comments inline - mostly around how it looks via rendering onto the docs page. Make sure to check how everything looks on the Quickstart section of the rendered docs built locally.

**Review by matthewhanson — APPROVED** [review]

**Pull request merged**